src/plopp/backends/matplotlib/line.py

```diff
@@ -221,6 +222,24 @@ def update(self, new_values: sc.DataArray):
                 )
             )
 
+    def format_coord(self, x: float, y: float) -> str:
+        """
+        Format the coordinates of the mouse pointer to show the value of the
+        data at that point.
+
+        Parameters
+        ----------
+        x:
+            The x coordinate of the mouse pointer.
+        y:
+            The y coordinate of the mouse pointer.
+        """
+        vx = sc.scalar(x, unit=self._canvas.xunit)
+        vy = sc.scalar(y, unit=self._canvas.yunit)
+        return (
+            f'{self._canvas.xdim}={scalar_to_string(vx)} : ' f'{scalar_to_string(vy)}'
```

Comment:
So this is not actually displaying the value, just the mouse-y? Then I think this `f(...): ...` syntax is very misleading (unlike in the 2-d case)

Reply:
So you are saying you don't like that it is not snapping to the closest value?
I guess this is difficult in the case of multiple lines. We don't really want one entry per line I think, because that may get very expensive with 20+ lines (which can easily happen when you use e.g. `sc.collapse`).

But that does raise a good point about the 2d images. Currently, although not common, you can have more than one image on the same axes (we prevent this in the `plot` function cf a discussion we had a while back), but it is still possible via the lower-level `FigImage`. If the `axformatter` is specified by the `Image` artist, when you add a new image it will override the formatter from the first image and only display values when you hover over the second image, not the first one.
So maybe I should have a global formatter on the canvas, and a mechanism where the artist register additional formatters.

For the 1d case, I am thinking that we should just display x and y and not try to find the closest value?

Reply:
> So you are saying you don't like that it is not snapping to the closest value?

No, I am saying we either have to snap to the closest value (and keep the display as it is), or change the display. As it is now the user may *think* that it displays the closest value, which is dangerous.

Reply:
I think i'd rather change the display to show the cursor x,y values. But I guess we still want the dim label and the units in there?
`time=40 s, 67 counts` would look weird though.
And we didn't want to use `x=40s, y=67 counts` because the horizontal axis may have `y` as a dimension, making things confusing...

Reply:
Hmm, maybe the dim label can be omitted here, since there is only one? `40 s, 67 counts`?

Reply:
Somehow I fear that even with this notation, sooner or later a user will think it is the actual value, not the mouse position. Can we make it even clearer?

Reply:
Can we display actual values, displaying a dict of values if there are multiple lines?

```
time:40 s -> (sample: 10 counts, background: 100 counts)
```

Reply:
I suppose that still does not solve the problem of which X value to use, since doing an interpolation seems wrong and too complicated.

Reply:
> Can we display actual values, displaying a dict of values if there are multiple lines?

We could, but not with 20 lines